app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java

```diff
@@ -234,48 +233,50 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         height = size.y;
 
         Bundle extras = getIntent().getExtras();
-        switch (extras.getString("who")) {
-            case "Half Wave Rectifier":
-                isHalfWaveRectifierExperiment = true;
-                if (scienceLab.isConnected()) {
-                    scienceLab.setSine1(5000);
-                }
-                break;
-            case "Full Wave Rectifier":
-                isFullWaveRectifierExperiment = true;
-                if (scienceLab.isConnected()) {
-                    scienceLab.setWaves(5000, 180, 5000);
-                }
-                break;
-            case "Diode Clipping Clamping":
-                isDiodeClippingClampingExperiment = true;
-                if (scienceLab.isConnected()) {
-                    scienceLab.setSine1(5000);
-                }
-                break;
-            case "Astable Multivibrator":
-                isAstableMultivibratorExperiment = true;
-                break;
-            case "Colpitts Oscillator":
-                isColpittsOscillatorExperiment = true;
-                break;
-            case "Phase Shift Oscillator":
-                isPhaseShiftOscillatorExperiment = true;
-                break;
-            case "Wien Bridge Oscillator":
-                isWienBridgeOscillatorExperiment = true;
-                break;
-            case "Monostable Multivibrator":
-                isMonostableMultivibratorExperiment = true;
-                break;
-            case "Speed of Sound":
-                isSpeedOfSoundExperiment = true;
-                break;
-            case "WaveGenerator":
-                //do nothing
-                break;
-            default:
-                break;
+        if (extras != null && extras.containsKey("who")) {
+            switch (extras.getString("who")) {
+                case "Half Wave Rectifier":
+                    isHalfWaveRectifierExperiment = true;
+                    if (scienceLab.isConnected()) {
+                        scienceLab.setSine1(5000);
+                    }
+                    break;
+                case "Full Wave Rectifier":
+                    isFullWaveRectifierExperiment = true;
+                    if (scienceLab.isConnected()) {
+                        scienceLab.setWaves(5000, 180, 5000);
+                    }
+                    break;
+                case "Diode Clipping Clamping":
```

Comment:
What if we remove these cases as experiments are no longer available?

Reply:
Yeah we can do that

Reply:
@CloudyPadmal I think it is better to do that in some other refractor because there is a lot of code to be removed

Reply:
I meant only the references made in this class